lib/resty/etcd/v3.lua

```diff
@@ -990,34 +1100,39 @@ function _M.timetolive(self, id, keys)
 end
 
 function _M.leases(self)
-    return _request_uri(self, "POST",
-                        choose_endpoint(self).full_prefix .. "/lease/leases")
+    local endpoint = choose_endpoint(self)
+    return _request_uri(self, endpoint, "POST",
+            endpoint.full_prefix .. "/lease/leases")
 end
 
 
 -- /version
 function _M.version(self)
-    return _request_uri(self, "GET",
-                        choose_endpoint(self).http_host .. "/version",
+    local endpoint = choose_endpoint(self)
+    return _request_uri(self, endpoint, "GET",
+            endpoint.http_host .. "/version",
                         nil, self.timeout)
 end
 
 -- /stats
 function _M.stats_leader(self)
-    return _request_uri(self, "GET",
-                        choose_endpoint(self).http_host .. "/v2/stats/leader",
+    local endpoint = choose_endpoint(self)
+    return _request_uri(self, endpoint, "GET",
+            endpoint.http_host .. "/v2/stats/leader",
                         nil, self.timeout)
 end
 
 function _M.stats_self(self)
-    return _request_uri(self, "GET",
-                        choose_endpoint(self).http_host .. "/v2/stats/self",
+    local endpoint = choose_endpoint(self)
+    return _request_uri(self, endpoint, "GET",
+            endpoint.http_host .. "/v2/stats/self",
```

Comment:
bad indentation

Reply:
code style: please fix similar points

Reply:
how about `return _request_uri(self, "GET", "/v2/stats/self", nil, self.timeout)`


```lua
local function _request_uri(self, method, path, opts, timeout, ignore_auth)
    local endpoint = choose_endpoint(self)
    local uri = endpoint.http_host .. path
    ......
```

Reply:
solve it